.github/workflows/python-build-test.yml

```diff
@@ -0,0 +1,45 @@
+name: Python Package Build Test
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.10', '3.11', '3.12']
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+    - name: Install uv
+      uses: astral-sh/setup-uv@6b9c6063abd6010835644d4c2e1bef4cf5cd0fca # v6.0.1
+      with:
+        python-version: ${{ matrix.python-version }}
+        activate-environment: true
+        version: 0.7.6
+
+    - name: Build Llama Stack package
+      run: |
+        uv build
+
+    - name: Install Llama Stack package
+      run: |
+        uv pip install dist/*.whl
+
+    - name: Verify Llama Stack package
```

Comment:
one failure case we have had is we forget to add a `__init__.py` every once in a while which results in the sub-folder not getting packaged. could we add a test for this?